website/server/controllers/api-v3/groups.js

```diff
@@ -718,4 +715,19 @@ api.inviteToGroup = {
   },
 };
 
+// Transitioning invitation storage to an array of objects (see issue #7792 for reference)
+function clearPartyInvitation(user, groupID) {
+  user.invitations.parties.forEach(function(party, index, partiesArr) {
```

Comment:
I think you can use the `removeFromArray` utility here. Then you won't even need to mark `user.invitations.parties` as modified.